### PR TITLE
Remove compat memset (eggdrop requires posix 2001)

### PR DESCRIPTION
**Pull request by michaelortmann**
Found by: vanosg
Patch by: michaelortmann
Fixes: #226 

One-line summary:
memset as part of issue 226

Additional description (if needed):
eggdrop depends on posix 2001. posix 2001 defines memset(). see http://www.unix.org/version3/inttables.pdf. this patch gets rid of src/compat/memset*. egg_bzero() and egg_memset() is left as is, but redefined to directly use memset() with a one-liner define in eggdrop.h. we only keep egg_memset() for backward compatibility to older modules, and could remove it from later eggdrop versions when api breakage becomes possible. while at it, i replaced some forgotten memset(dest, 0, n) with egg_bzero() and removed one register keyword in compress.c. maybe we should open an issue calling for removal of most of register keywords within eggdrop. modern compilers are good at register allocation.

Test cases demonstrating functionality (if applicable):
i added compress mod to eggdrop.conf and started eggdrop as minimal test. so far so good.

### CODE REVIEW

File: src/modules.c [hunk 1]
```diff
@@ -543,7 +543,7 @@ Function global_table[] = {
   /* 252 - 255 */
   (Function) egg_snprintf,
   (Function) egg_vsnprintf,
-  (Function) egg_memset,
+  (Function) 0,                   /* was egg_memset -- use memset() instead */
```
Comment (Cizzle) [inline src/modules.c]: Comment should rather point to egg_bzero?

**Comment by Cizzle** [comment]
LGTM apart from the nitpick comment. Merger should be cautious not to automatically close #226.

File: src/modules.c (hunk 1 above)
Comment (michaelortmann) [reply src/modules.c]: its done

**Comment by Cizzle** [comment]
LGTM²

**Comment by michaelortmann** [comment]
removing a function from table in src/modules.c looks like module api change? maybe delay for 1.9.0 instead of 1.8.x?

**Comment by vanosg** [comment]
Agree - however small the chance is, some other module could be calling egg_memset. Pending any objections, I'll move this to 1.9.0.

Aside from that, nice modernization to codebase. Always fun to see the code size actually shrink after a PR!

File: src/modules.c [hunk 2]
```diff
@@ -543,7 +543,7 @@ Function global_table[] = {
   /* 252 - 255 */
   (Function) egg_snprintf,
   (Function) egg_vsnprintf,
-  (Function) egg_memset,
+  (Function) 0,                   /* was egg_memset -- use memset() or egg_bzero() instead */
```
Comment (thommey) [inline src/modules.c:546]: could we export memset if it has the same signature at this point to avoid breaking modules?

**Comment by thommey** [comment]
Did you manually remove memset.h from Makefile.in or did you run "misc/makedepend"? If not, misc/makedepend would be preferred so it becomes a no-op when we run it in the future (line breaks etc.)

File: src/modules.c (hunk 2 above)
Comment (michaelortmann) [reply src/modules.c:546]: yes. done.
src/eggdrop.h
#define egg_memset(dest, c, len) memset(dest, c, len)
i did a quick test, and egg_memset was once again useable in module.

**Comment by michaelortmann** [comment]
now i also did run misc/makedepend.

**Comment by michaelortmann** [comment]
because egg_memset is once again useable by modules, i hereby request milestone 1.8.4 instead of 1.9.0.

**Review by thommey — APPROVED** [review]
LGTM

**Comment by vanosg** [comment]
builf

**Comment by vanosg** [comment]
build

**Pull request merged**